plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -1026,12 +1040,18 @@ def save_fileless_units(self, file_handle, tag, process_func, mutable_type=False
                             continue
                         else:
                             raise
-                    model.save()
+                    if has_changed:
+                        model.save()
                 else:
                     # make sure the associate_unit call gets the existing unit
                     model = existing_unit
+                    # FIXME: Should update the mutable unit
```

Comment:
Just making sure this doesn't get overlooked.

Reply:
@hao-yu

Reply:
@dralley As far as I can see, this is always a bug in Pulp tha it doesn't update the existing comps units so I added this "FIXME" here just to let everyone know about it and correct me in case I am wrong. I didn't try to fix it because it is not related to the purpose of this patch.